cmd/soroban-cli/src/commands/contract/mod.rs

```diff
@@ -165,3 +167,8 @@ pub enum SpecOutput {
     /// Pretty print of contract spec entries
     Docs,
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct AliasData {
+    id: String,
+}
```

Comment:
It would require creating a new Error type (perhaps it's worth moving to a separate file under `contract/config`, but then you could move your validation logic to parsing
```
impl FromStr for AliasData {

   type Error = Error; // The local error type which will include `Error::InvalidAliasFormat`
   
    fn from_str(s: &str) -> Result<Self, Self::Error> {
        let regex = Regex::new(r"^[a-zA-Z0-9_-]{1,30}$")?; // Here you should add the regex error (also there should be a way to have this done at compile time to avoid a runtime check on the validity of the regex)
        let alias = s.to_owned(); // My personal preferred way of going `&str` to `String`, as `String` is the owned version of the reference.
        if regex.is_match(&alias) {
            Ok(Self { id: alias } )
        } else {
            Err(Error::InvalidAliasFormat { alias })
        }
    }

}
```
See https://doc.rust-lang.org/std/str/trait.FromStr.html
There is then the `parse` method on `&str` which generic but requires that the `T` implements `FromStr`.

Then you would have above:

```rust
        let payload = AliasData {
            id: contract.into(),
        };
```
Becomes
```rust
        let payload: AliasData = contract.parse()?;
```
The compiler can then infer the type for `parse` so it's the same as 
```rust
        let payload = contract.parse::<AliasData>()?;
```

This article called [Parse, don't Validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/) uses haskell to explain it, but the TL;DR is that by parsing a type, it pushing the validation of the type to it's creation avoid footguns and creating a clear error.

Reply:
Oops just realized that here `id` is a contract id! In this case you would want to move the contract_id parsing logic here. 
```rust
impl FromStr for AliasData {

   type Error = Error; // The local error type which will include `Error::InvalidAliasFormat`
   
    fn from_str(s: &str) -> Result<Self, Self::Error> {
        let id = soroban_spec_tools::utils::contract_id_from_str(&contract_id).map_err(|e| Error::CannotParseContractId(contract_id.clone(), e))?;
        Ok(Self{ id })
    }

}
```


Then you could create a method 

```rust
impl AliasData {
    fn write_to_file(&self, alias: &str, dir: &Path) -> Result<(), Error> {
        // Check the alias here
    }
}

Reply:
Also note that the `FromStr` wouldn't make sense if we changed the struct to have a Map.  So I would instead create a `new` method that returns `Result<Self, Error>`.

Reply:
I don't think we can follow this pattern, as we need to validate the alias before we make the call to RPC. Otherwise you'd need to fail on a operation that has already succeeded.